src/components/AmountInput.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div>
+    <b-row>
+      <b-col>
+        <b-form-input
+          id="input-amount"
+          v-model="amount"
+          :type="'number'"
+          placeholder="Enter amount"
+          aria-describedby="input-live-help input-live-feedback"
+          @keyup="validateAmount">
+        </b-form-input>
+        <p v-if="errorMsg">{{ errorMsg }}</p>
+      </b-col>
+      <b-col sm="4">
+        <b-button variant="outline-primary" @click="setAllAmount">All ({{ state.plasma.tokenSelected.toUpperCase() }})</b-button>
+      </b-col>
+    </b-row>
+    
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Prop, Component, Watch } from "vue-property-decorator"
+import { DashboardState } from '@/types';
+
+@Component
+
+export default class AmountInput extends Vue {
+  @Prop(Number) value!: number // v-model is it accepts a value prop and emit an input event.
+  @Prop(Number) min!: number
+  @Prop(Number) max!: number
+  @Prop({default: true}) round!: boolean
+
+  // State declaration
+  amount: any = this.value
+  errorMsg: string = ""
+
+  // Call this function when amount changed
+  @Watch("amount")
+  onAmountChanged(newVal, oldVal) {
+    this.$emit("input", Number(this.amount))
+  }
+
+  // Set default amount when select another token
+  @Watch("state.plasma.tokenSelected")
+  setDefaultAmount(newVal, oldVal){
+    this.amount = 0
+  }
+
+  validateAmount() {
```

Comment:
Although we’re using input type number the value return can be a strong and it should be handled

Reply:
strong or string?
should we emit an amount to parent component as integer type or BN type?